bufr2geojson/__init__.py

```diff
@@ -521,26 +538,27 @@ def get_code_value(self, fxxyyy: str, code: int) -> str:
 
         :returns: string representation of coded value
         """
-        table = self.code_table.loc[(self.code_table["FXY"] == fxxyyy), ]
-        decoded = table.loc[table["CodeFigure"] == code, ]
-        decoded.reset_index(drop=True, inplace=True)
-        if len(decoded) == 1:
-            decoded = decoded.EntryName_en[0]
-        else:
-            assert len(decoded) == 0
-            decoded = None
+
+        table = {k: v for k, v in self.code_table.items() if k == fxxyyy}
+        decoded = None
+        if table[fxxyyy]['CodeFigure'] == code:
```

Comment:
Here we need to iterate over the rows to find the one where the CodeFigure matches the code we are looking for.